esmvaltool/preprocessor/_multimodel.py

```diff
@@ -13,27 +31,162 @@
 logger = logging.getLogger(__name__)
 
 
-def _put_in_cube(stats, stats_name, ncfiles, fname):
-    """quick cube building"""
-    stats_cube = iris.cube.Cube(stats, long_name=stats_name)
-    stats_cube.attributes['_filename'] = fname
-    stats_cube.attributes['NCfiles'] = str(ncfiles)
+def _plev_fix(dataset, pl_idx):
+    """Extract valid plev data
+
+    this function takes care of situations
+    in which certain plevs are completely
+    masked due to unavailable interpolation
+    boundaries.
+    """
+    if np.ma.is_masked(dataset) is True:
+        # keep only the valid plevs
+        if not np.all(dataset.mask[pl_idx]):
+            statj = np.ma.array(dataset[pl_idx],
+                                mask=dataset.mask[pl_idx])
+        else:
+            logger.debug('All vals in plev are masked, ignoring.')
+            statj = None
+    else:
+        mask = np.zeros_like(dataset[pl_idx], bool)
+        statj = np.ma.array(dataset[pl_idx], mask=mask)
+
+    return statj
+
+
+def _compute_statistic(datas, name):
+    """Compute multimodel statistic"""
+    datas = np.ma.array(datas)
+    statistic = datas[0]
+
+    if name == 'median':
+        statistic_function = np.ma.median
+    elif name == 'mean':
+        statistic_function = np.ma.mean
+    else:
+        raise NotImplementedError
+
+    # no plevs
+    if len(datas[0].shape) < 3:
+        statistic = statistic_function(datas, axis=0)
+        return statistic
+
+    # plevs
+    for j in range(statistic.shape[0]):
```

Comment:
Try to call `_plev_fix` only once per pressure level, because this is inefficient. You could also consider using some fancy array slicing (similar to how it's done in the `mask_fillvalues` function) instead of calling `_plev_fix`.

Reply:
ok, calling it once now, not in the nicest way but that'll do for now; no change in memory intake tho

Reply:
You're still calling it twice in the code below, on line 89 and on line 91. Indeed this should not make a difference in memory use, but in computation time, because every time you call that function you check if all elements are masked.